components/filter/filter.js

```diff
@@ -415,9 +419,13 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 						?hidden="${item.hidden}"
 						key="${item.key}"
 						label="${item.text}"
+						number="${item.count}"
 						selectable
 						?selected="${item.selected}">
-						<div class="d2l-filter-dimension-set-value-text d2l-body-compact">${item.text}</div>
+						<div class="d2l-filter-dimension-set-value d2l-body-compact">
+							<div class="d2l-filter-dimension-set-value-text">${item.text}</div>
+							${item.count !== undefined ? html`<div class="d2l-body-small">(${formatNumber(item.count)})</div>` : nothing}
```

Comment:
I have tried testing this out using the demo pages and when I pass a string into "count" the filter dropdown just breaks and does not render at all even if I have the `Number.isNaN(item.count)` check.

Reply:
Just updating here with what I already mentioned in Slack. I think this can be done a couple different ways.

My preference is to validate closer to the assignment on `d2l-filter-dimension-set-value` with something like...
```javascript
get count() {
	return this._count;
}

set count(val) {
	if (Number.isNaN(val)) val = undefined;
	const oldVal = this._count;
	if (oldVal !== val) {
		this._count = val;
		this.requestUpdate('count', oldVal);
	}
}
```

But validating on usage in `d2l-filter` can in fact work as well:
```javascript
${(item.count !== undefined && !Number.isNaN(item.count)) ? html`<div class="d2l-body-small">(${formatNumber(item.count)})</div>` : nothing}
```

I've tested both these approaches. It's a worthwhile check. I would not expect consumers to actually set a string value purposefully, but if the consumer is provided an invalid count, we'd like the UI component to not produce JavaScript errors or the invalid value.

Reply:
Made the change for an additional check on string inputs for count.